src/commons/api/misc.rs

```diff
@@ -0,0 +1,31 @@
+//! Miscellaneous types related to the API.
+
+use std::fmt;
+use serde::ser::{Serialize, Serializer, SerializeStruct};
+
+
+//------------ Success -------------------------------------------------------
+
+/// An empty, successful API response.
+///
+/// This type needs to be used instead of `()` to make conversion into
+/// [`Report`][crate::client::report::Report] work.
+#[derive(Clone, Copy, Debug)]
+pub struct Success;
+
+impl fmt::Display for Success {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_str("Ok")
+    }
+}
+
+impl Serialize for Success {
+    fn serialize<S: Serializer>(
+        &self, serializer: S
+    ) -> Result<S::Ok, S::Error> {
+        let mut serializer = serializer.serialize_struct( "UnitSuccess", 1)?;
```

Comment:
Why "UnitSuccess" and not "Success"?

Reply:
That’s probably leftover from an earlier name of the struct. I’ll fix it.